boards/launchxl/src/uart_echo.rs

```diff
@@ -123,10 +72,15 @@ impl<UTx: 'static + uart::Transmit<'static>, URx: 'static + uart::Receive<'stati
         }
     }
 
-    pub fn initialize(&self) {
-        self.rx_buf.take().map(|buf| {
-            self.uart_rx.receive_buffer(buf, MAX_PAYLOAD);
-        });
+    pub fn initialize(&self, word_mode: bool) {
+        if word_mode {
+            debug!("putting self as receive word");
+            self.uart_rx.receive_word();
+        } else {
+            self.rx_buf.take().map(|buf| {
+                self.uart_rx.receive_buffer(buf, MAX_PAYLOAD);
+            });
+        }
```

Comment:
Unrelated to this PR but you will never catch if the the `Option` is none here would probably be good to return some status or add a debug message!